systems/subscriber/subscriber-registry/pkg/server/subscriber.go

```
@@ -0,0 +1,288 @@
+		return nil, err
+	}
+
+	timestamp := &timestamppb.Timestamp{
+		Seconds: req.DateOfBirth.GetSeconds(),
+		Nanos:   req.DateOfBirth.GetNanos(),
+	}
+
+	birthday := timestamp.AsTime()
+
+	subscriber := &db.Subscriber{
+		OrgID:orgID_uuid,
+		SubscriberID:          subscriberID_uuid,
+		FirstName:             req.GetFirstName(),
+		LastName:              req.GetLastName(),
+		NetworkID:             networkID_uuid,
+		Email:                 req.GetEmail(),
+		PhoneNumber:           req.GetPhoneNumber(),
+		Gender:                req.GetGender(),
+		Address:               req.GetAddress(),
+		ProofOfIdentification: req.GetProofOfIdentification(),
+		DOB:                   birthday,
+		IdSerial:              req.GetIdSerial(),
+	}
+	err = s.subscriberRepo.Add(subscriber)
+	if err != nil {
+		logrus.Error("error while adding subscriber" + err.Error())
+		return nil, grpc.SqlErrorToGrpc(err, "subscriber")
+	}
+
+	return &pb.AddSubscriberResponse{
+		Subscriber: &pb.Subscriber{
+			OrgID: orgID_uuid.String(),
+			SubscriberID:          subscriberID_uuid.String(),
+			FirstName:             req.GetFirstName(),
+			LastName:              req.GetLastName(),
+			NetworkID:             networkID_uuid.String(),
+			Email:                 req.GetEmail(),
+			PhoneNumber:           req.GetPhoneNumber(),
+			Gender:                req.GetGender(),
+			Address:               req.GetAddress(),
+			ProofOfIdentification: req.GetProofOfIdentification(),
+			DateOfBirth:           birthday.GoString(),
+			IdSerial:              req.GetIdSerial()},
+	}, nil
+
+}
+
+func (s *SubcriberServer) Delete(ctx context.Context, req *pb.DeleteSubscriberRequest) (*pb.DeleteSubscriberResponse, error) {
+	subscriberID := req.GetSubscriberID()
+	if subscriberID == "" {
+		return nil, status.Errorf(codes.InvalidArgument, "subscriberID must not be empty")
+	}
+
+	if err := ctx.Err(); err != nil {
+		return nil, err
+	}
+	subscriberIDUUID, err := uuid.FromString(subscriberID)
```

1) subscriberID or subscriberUUID
2) We never swallow errors in Go. Please always handle any resulting error from any given operation.

---

Addressed.